sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/InstrumentationRegistryHelper.java

```diff
@@ -6,22 +6,59 @@
 package com.microsoft.appcenter.utils;
 
 import android.os.Bundle;
-import android.support.test.InstrumentationRegistry;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
 
 /**
  * Wraps InstrumentationRegistry class to enable mocking in unit tests.
  */
 public class InstrumentationRegistryHelper {
+    private static List<String> locations = Arrays.asList("androidx.test.platform.app.InstrumentationRegistry",
+            "androidx.test.InstrumentationRegistry",
+            "android.support.test.InstrumentationRegistry");
 
     /**
      * Get the instrumentation arguments from the InstrumentationRegistry. Wrapper exists for unit
      * tests.
      *
      * @return the instrumentation arguments.
      * @throws LinkageError          if the class, method is not found or does not match, typically no test dependencies in release.
-     * @throws IllegalStateException if no argument Bundle has been registered.
      */
     public static Bundle getArguments() throws LinkageError, IllegalStateException {
-        return InstrumentationRegistry.getArguments();
+        Iterator<String> iterator = locations.iterator();
+        while (iterator.hasNext()) {
+            String location = iterator.next();
+
+            try {
+                Class<?> aClass = Class.forName(location);
+                Method getArguments = aClass.getMethod("getArguments", (Class[]) null);
+                return (Bundle) getArguments.invoke(null, (Object[]) null);
+                // We need to support api level 8 and up, so we cannot combine catches into one
+            } catch (IllegalStateException e) {
+                // Ignore unless last item
+                if (!iterator.hasNext()) {
+                    throw e;
+                }
+            } catch (LinkageError e) {
+                // Ignore unless last item
+                if (!iterator.hasNext()) {
+                    throw e;
+                }
+            } catch (ClassNotFoundException e) {
```

Comment:
For ignoring we name it `ignore` or `ignored` that will disable the code analysis warning. Also just use Exception here for reflection we do that.